### PR TITLE
Replace node_package with relx

### PR DESCRIPTION
**Pull request by sanmiguel**
This replaces `node_package` packaging with `relx`.

Currently this strips away some configurable stuff, which I'll work on restoring now while I tidy up the outmoded artifacts.

One major reason for this change is that `node_package` doesn't seem to have a way to create a package that will start without distribution. Without this, the scheduler itself connects to EPMD which we can't rely on running (outside of DC/OS - more on that later).

By explicitly using an empty `vm.args` file, the scheduler is started without the `-sname riak_mesos_scheduler` arg, which is what ultimately results in attempting to connect to EPMD.

The downside of not connecting to EPMD is that you lose the ability to attach (via remsh) to a running scheduler, which will make debugging in live environments harder. However, since the scheduler can be relatively freely restarted and replaced in live environments, it's feasible to provide a scheduler artifact with a vm.args that starts erlang distribution and therefore allows remsh, and restart the scheduler.

To do this, one simply removes the `vm_args` configuration from `relx.config` and builds a new release.

~~Still TODO~~:
 - [x] sys.config
 - [x] remove cuttlefish/node_package artifacts

### CODE REVIEW

**Comment by sanmiguel** [comment]
As promised, more on EPMD in DCOS:

A core DCOS component is `minuteman`: a distributed erlang load balancer. To support this, another DCOS component is an EPMD instance running on a non-standard port. If this EPMD instance runs on every mesos-agent, and that port is available by interrogating DCOS configuration endpoints, we should be able to piggyback on that EPMD instance and use the scheduler in a distributed mode allowing remote-shell access (easier for debugging). To properly allow this we'll need to make further changes (beyond the scope of this PR) to ensure we start the scheduler with a more unique name than `riak_mesos_scheduler`.

**Comment by sanmiguel** [comment]
Oh - also NB: `make rel` no longer depends on `make clean-deps` and therefore is no longer desctructive. If you really want to delete and re-fetch all the dependencies again, call `make clean-deps` manually.

**Comment by paegun** [comment]
+1 7a8e558

I considered whether some of this PR's commentary belongs in the README, but feel it is:
* covered by relx -v- nodetool knowledge
* in commentary/documentation throughout the project

Thank you for bringing that knowledge into the PR commentary for easier review.

File: rebar.config.lock [hunk 1]
```diff
@@ -88,14 +76,26 @@
        {riakc,".*",
               {git,"git://github.com/basho/riak-erlang-client",
                    "802119823bc94459e1e4ac0d747b26e0ae1f14cc"}},
+       {node_package,".*",
```
Comment (drewkerrigan) [inline rebar.config.lock]: Something seems to be keeping this around?

**Comment by drewkerrigan** [comment]
👍 LGTM

**Pull request merged**